Mathlib/RingTheory/Support.lean

```diff
@@ -201,4 +211,35 @@ lemma LocalizedModule.exists_subsingleton_away [Module.Finite R M] (p : Ideal R)
   exact ⟨f, by simpa using hf', subsingleton_iff.mpr
     fun m ↦ ⟨f, Submonoid.mem_powers f, Module.mem_annihilator.mp hf _⟩⟩
 
+/-- `Supp(M/IM) = Supp(M) ∩ Z(I)`. -/
+@[stacks 00L3 "(1)"]
+theorem Module.support_quotient (I : Ideal R) :
+    support R (M ⧸ (I • ⊤ : Submodule R M)) = support R M ∩ zeroLocus I := by
+  apply subset_antisymm
+  · refine Set.subset_inter ?_ ?_
+    · exact Module.support_subset_of_surjective _ (Submodule.mkQ_surjective _)
+    · rw [support_eq_zeroLocus]
+      apply PrimeSpectrum.zeroLocus_anti_mono_ideal
+      rw [Submodule.annihilator_quotient]
+      exact fun x hx ↦ Submodule.mem_colon.mpr fun p ↦ Submodule.smul_mem_smul hx
+  · rintro p ⟨hp₁, hp₂⟩
+    rw [Module.mem_support_iff] at hp₁ ⊢
+    let Rₚ := Localization.AtPrime p.asIdeal
+    let Mₚ := LocalizedModule p.asIdeal.primeCompl M
+    set Mₚ' := LocalizedModule p.asIdeal.primeCompl (M ⧸ (I • ⊤ : Submodule R M))
+    let Mₚ'' := Mₚ ⧸ I.map (algebraMap R Rₚ) • (⊤ : Submodule Rₚ Mₚ)
+    let e : Mₚ' ≃ₗ[Rₚ] Mₚ'' := (localizedQuotientEquiv _ _).symm ≪≫ₗ
+      Submodule.quotEquivOfEq _ _ (by rw [Submodule.localized,
+        Submodule.localized'_smul, Ideal.localized'_eq_map, Submodule.localized'_top])
+    have : Module.Finite Rₚ Mₚ :=
```

Comment:
Is it possible to have those as instances? `Module.Finite.of_isLocalizedModule` cannot be, but I think having the specific version is a good idea. Similarly to what I am doing in #21522.